onnxruntime/core/graph/graph.cc

```diff
@@ -4046,6 +4070,210 @@ Status Graph::AddConstantProtoAsInitializer(const ONNX_NAMESPACE::NodeProto& nod
   return Status::OK();
 }
 
+static void RenameSubgraphDependentNames(const InlinedHashMap<std::string, std::string>& name_mapping,
+                                         ONNX_NAMESPACE::GraphProto& graph_proto) {
+  for (auto& node : *graph_proto.mutable_node()) {
+    for (auto& attr : *node.mutable_attribute()) {
+      if (attr.has_g()) {
+        RenameSubgraphDependentNames(name_mapping, *attr.mutable_g());
+      }
+
+      for (auto& subgraph : *attr.mutable_graphs()) {
+        RenameSubgraphDependentNames(name_mapping, subgraph);
+      }
+    }
+
+    for (auto& input : *node.mutable_input()) {
+      auto hit = name_mapping.find(input);
+      if (hit != name_mapping.cend()) {
+        input = hit->second;
+      }
+    }
+  }
+}
+
+static void RenameNodeAttributesSubgraphDependentNames(const InlinedHashMap<std::string, std::string>& name_mapping,
```

Comment:
I wonder if either this can be dropped or if the duplicated lines in the previous function (lines 4076 onwards) be replaced by a call to this function?

Reply:
This is being re-worked.